src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -304,7 +304,14 @@ private static string GetSize(long length)
         private static Option ProvidersOption() =>
             new Option(
                 alias: "--providers",
-                description: @"A list of EventPipe providers to be enabled. This is in the form 'Provider[,Provider]', where Provider is in the form: 'KnownProviderName[:Flags[:Level][:KeyValueArgs]]', and KeyValueArgs is in the form: '[key1=value1][;key2=value2]'. These providers are in addition to any providers implied by the --profile argument. If there is any discrepancy for a particular provider, the configuration here takes precedence over the implicit configuration from the profile.",
+                description: @"A comma delimitted list of EventPipe providers to be enabled. This is in the form 'Provider[,Provider]'," +
+                             @"where Provider is in the form: 'KnownProviderName[:Flags[:Level][:KeyValueArgs]]', and KeyValueArgs is in the form: " +
+                             @"'[key1=value1][;key2=value2]'.  Values in KeyValueArgs that contain ';' or '=' characters need to be surrounded by '""', " +
+                             @"e.g., FilterAndPayloadSpecs=""MyProvider/MyEvent:-Prop1=Prop1;Prop2=Prop2.A.B;"".  Depending on your shell, you may need to " +
+                             @"escape the '""' characters and/or surround the entire provider specification in quotes, e.g., " +
+                             @"--providers 'Name:1:1:FilterSpec=\""MyProvider/MyEvent:-Prop1=Prop1;Prop2=Prop2.A.B;\""'. These providers are in " +
```

Comment:
1. Use a real provider and use a hex value for the keywords field.
2. Get rid of this nested of brackets `KnownProviderName[:Flags[:Level][:KeyValueArgs]]`, make it `ProviderName:[Flags]:[Level]:[KeyValueArgs]`.  I don't understand why it's nested.

Reply:
I'll update the example to use an actual provider.

I think they are nested because you aren't required to specify the subsequent `:`'s, e.g., `KnownProviderName:0x1` is valid.  That being said, I think the nesting is actually incorrect.  It _should_ be `ProviderName[:[Keywords][:[Level][:[KeyValueArgs]]]]` since you must put the colons in for `Keywords` and `Level` to specify any `KeyValueArgs` even if you don't specify values for `Keywords` and `Level`, e.g., `ProviderName:::Key=Value`.

Reply:
I don’t think it’s clear at all. The nesting makes it more confusing

Reply:
I do see @davidfowl's point that this is really hard to read. That being said, I do think @josalem's notation is actually the correct descriptor of the argument. @noahfalk Do you have any thoughts on what we should do about this? Our descriptor of the argument is hard to read, but the easier-to-read version is technically an incorrect descriptor of what we accept.

Reply:
One approach we could take is to show an easier-to-read version of the spec in the help text and have the full specification in the documentation.  S.CommandLine seems to remove all custom formatting from help text so it is easier to keep the help text short and direct people to documentation.

Reply:
We could give it to 5 people and see if they can figure it out 😄 (without the example)

Reply:
> Do you have any thoughts on what we should do about this? Our descriptor of the argument is hard to read, but the easier-to-read version is technically an incorrect descriptor of what we accept.

Looking at the current implementation name:[flags]:[level]:[filterParams] is a valid subset of syntax what we would accept. I think its fine to use simpler approximations in small text descriptions and then put more rigorous descriptions elsewhere for the 1% of users who need them. We could take a bunch of the text out of line if we have a dedicated command or help topic that describes provider names in detail. For example:

```
--providers <list-of-comma-separated-providers>    A provider list is bla bla bla. For more 
                                                   details on provider formatting syntax use
                                                   'dotnet-trace help provider_format'

```